pkg/controller/enterprisesearch/enterprisesearch_controller_test.go

```diff
@@ -277,3 +277,245 @@ func TestReconcileEnterpriseSearch_Reconcile_Create_Update_Resources(t *testing.
+				ReconcilerStatus: commonv1.ReconcilerStatus{
+					AvailableNodes: 4,
+				},
+				Health:          "green",
+				ExternalService: "http-service",
+			},
+			wantStatusUpdateCalled: true,
+		},
+		{
+			name: "don't do a status update if not necessary",
+			ent: entv1beta1.EnterpriseSearch{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "ent"},
+				Status: entv1beta1.EnterpriseSearchStatus{
+					ReconcilerStatus: commonv1.ReconcilerStatus{
+						AvailableNodes: 3,
+					},
+					Health:          "green",
+					ExternalService: "http-service",
+				}},
+			deploy: appsv1.Deployment{Status: appsv1.DeploymentStatus{
+				AvailableReplicas: 3,
+				Conditions: []appsv1.DeploymentCondition{
+					{
+						Type:   appsv1.DeploymentAvailable,
+						Status: corev1.ConditionTrue,
+					},
+				},
+			}},
+			svcName: "http-service",
+			wantStatus: entv1beta1.EnterpriseSearchStatus{
+				ReconcilerStatus: commonv1.ReconcilerStatus{
+					AvailableNodes: 3,
+				},
+				Health:          "green",
+				ExternalService: "http-service",
+			},
+			wantStatusUpdateCalled: false,
+		},
+		{
+			name: "emit an event when health goes from green to red",
+			ent: entv1beta1.EnterpriseSearch{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "ent"},
+				Status: entv1beta1.EnterpriseSearchStatus{
+					ReconcilerStatus: commonv1.ReconcilerStatus{
+						AvailableNodes: 3,
+					},
+					Health:          "green",
+					ExternalService: "http-service",
+				}},
+			deploy: appsv1.Deployment{Status: appsv1.DeploymentStatus{
+				AvailableReplicas: 3,
+				Conditions: []appsv1.DeploymentCondition{
+					{
+						Type:   appsv1.DeploymentAvailable,
+						Status: corev1.ConditionFalse,
+					},
+				},
+			}},
+			svcName: "http-service",
+			wantStatus: entv1beta1.EnterpriseSearchStatus{
+				ReconcilerStatus: commonv1.ReconcilerStatus{
+					AvailableNodes: 3,
+				},
+				Health:          "red",
+				ExternalService: "http-service",
+			},
+			wantEvent:              true,
+			wantStatusUpdateCalled: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &fakeClientStatusCall{Client: k8s.WrappedFakeClient(&tt.ent)}
+			fakeRecorder := record.NewFakeRecorder(10)
+			r := &ReconcileEnterpriseSearch{
+				Client:   c,
+				recorder: fakeRecorder,
+			}
+			err := r.updateStatus(tt.ent, tt.deploy, tt.svcName)
+			require.NoError(t, err)
+
+			require.Equal(t, tt.wantStatusUpdateCalled, c.updateCalled)
+
+			var updatedEnt entv1beta1.EnterpriseSearch
+			err = c.Get(k8s.ExtractNamespacedName(&tt.ent), &updatedEnt)
+			require.NoError(t, err)
+			require.Equal(t, tt.wantStatus, updatedEnt.Status)
+
+			if tt.wantEvent {
+				<-fakeRecorder.Events
+			} else {
+				// no event expected
+				select {
+				case e := <-fakeRecorder.Events:
+					require.Fail(t, "no event expected but got one", "event", e)
+				default:
+					// ok
+				}
+			}
+		})
+	}
+}
+
+//
+//func TestReconcileEnterpriseSearch_noUpdate(t *testing.T) {
```

Comment:
I think this one is part of TestReconcileEnterpriseSearch_updateStatus now ?

Reply:
🤦